x/wasm/keeper/keeper_test.go

```diff
@@ -1309,3 +1309,90 @@ func TestInitializePinnedCodes(t *testing.T) {
 		assert.Equal(t, exp, capturedChecksums[i])
 	}
 }
+
+func TestNewDefaultWasmVMContractResponseHandler(t *testing.T) {
+	noopDMsgs := func(ctx sdk.Context, contractAddr sdk.AccAddress, ibcPort string, msgs []wasmvmtypes.CosmosMsg) error {
+		return nil
+	}
+
+	specs := map[string]struct {
+		srcData []byte
+		setup   func(m *wasmtesting.MockMsgDispatcher)
+		expErr  bool
+		expData []byte
+	}{
+		"submessage overwrites result when set": {
+			srcData: []byte("otherData"),
+			setup: func(m *wasmtesting.MockMsgDispatcher) {
+				m.DispatchSubmessagesFn = func(ctx sdk.Context, contractAddr sdk.AccAddress, ibcPort string, msgs []wasmvmtypes.SubMsg) ([]byte, error) {
+					return []byte("mySubMsgData"), nil
+				}
+				m.DispatchMessagesFn = noopDMsgs
+			},
+			expErr:  false,
+			expData: []byte("mySubMsgData"),
+		},
+		"submessage overwrites result when empty": {
+			srcData: []byte("otherData"),
+			setup: func(m *wasmtesting.MockMsgDispatcher) {
+				m.DispatchSubmessagesFn = func(ctx sdk.Context, contractAddr sdk.AccAddress, ibcPort string, msgs []wasmvmtypes.SubMsg) ([]byte, error) {
+					return []byte(""), nil
```

Comment:
We need to look closely at the FFI boundary and see if we have that distinction between `[]byte("")` and `nil` available